test_simulation.cpp

```diff
@@ -171,6 +172,164 @@ int test_with_reduced_stochasticity()
+        return 1;
+    }
+    if (infected != expected_infected) {
+        cout << "Deterministic Kernel Cauchy: infected (actual, expected):\n" << infected << "  !=\n" << expected_infected << "\n";
+        return 1;
+    }
+    if (mortality_tracker != expected_mortality_tracker) {
+        cout << "Deterministic Kernel Cauchy: mortality tracker (actual, expected):\n" << mortality_tracker << "  !=\n" << expected_mortality_tracker << "\n";
+        return 1;
+    }
+    cout << "Deterministic Kernel Cauchy: no errors\n";
+    
+    infected = {{5, 0, 0}, {0, 5, 0}, {0, 0, 2}};
+    mortality_tracker = {{0, 0, 0}, {0, 0, 0}, {0, 0, 0}};
+    susceptible = {{10, 20, 9}, {14, 15, 0}, {3, 0, 2}};
+    total_hosts = susceptible;
+    movements = {{0, 0, 1, 1, 2}, {0, 1, 0, 0, 3}, {0, 1, 1, 0, 2}};
+    movement_schedule = {1, 1};
+	std::vector<std::tuple<int, int>> outside_dispersers2;
+    expected_mortality_tracker = {{10, 0, 0}, {0, 10, 0}, {0, 0, 2}};
+    expected_infected = {{15, 0, 0}, {0, 15, 0}, {0, 0, 4}};
+    dispersers(infected.rows(), infected.cols());
+    
+    // Exponential
+    Simulation<Raster<int>, Raster<double>> s2(
+                42,
+                infected.rows(),
+                infected.cols(),
+                model_type_from_string("SI"),
+                0,
+                generate_stochasticity,
+                establishment_stochasticity
+                );
+    s2.generate(dispersers, infected, weather, weather_coefficient, reproductive_rate);
+    expected_dispersers = reproductive_rate * infected;
+    if (dispersers != expected_dispersers) {
+        cout << "Deterministic Kernel Exponential: dispersers (actual, expected):\n" << dispersers << "  !=\n" << expected_dispersers << "\n";
+        return 1;
+    }
+    RadialDispersalKernel deterministicKernel2(30, 30, DispersalKernelType::Exponential, 1.0, Direction::None, 0.0, true, dispersers, 0.99, 0.0);
+   
+    s2.disperse(dispersers, susceptible, infected,
+                        mortality_tracker, total_hosts,
+                        outside_dispersers2, weather, weather_coefficient,
+                        deterministicKernel2, establishment_probability);
+    if (outside_dispersers2.size() != 0) {
+        cout << "Deterministic Kernel Exponential: There are outside_dispersers (" << outside_dispersers2.size() << ") but there should be 0\n";
+        return 1;
+    }
+    if (infected != expected_infected) {
+        cout << "Deterministic Kernel Exponential: infected (actual, expected):\n" << infected << "  !=\n" << expected_infected << "\n";
+        return 1;
+    }
+    if (mortality_tracker != expected_mortality_tracker) {
+        cout << "Deterministic Kernel Exponential: mortality tracker (actual, expected):\n" << mortality_tracker << "  !=\n" << expected_mortality_tracker << "\n";
+        return 1;
+    }
+    cout << "Deterministic Kernel Exponential: no errors\n";
+    
+    // testing cauchy pdf & cdf
```

Comment:
This is separate from the rest, so make it a separate function.

In fact, it would be ideal if this would be in a separate file. However, there is couple of things which need to be done for that (see Makefile), so I can do it later.

Reply:
I moved this to a different function, but can move it to a different file later if you would like.

Reply:
Well, that would be actually quite useful because the test_simulation.cpp is getting long. There is no doc for adding more test files, so let me know if I should clarify something.

Reply:
I created a new test file test_deterministic.cpp and I moved the deterministic kernel tests to that file. I also updated the makefile.